packages/react/src/date-picker/reactDayPickerStyles.ts

```diff
@@ -4,140 +4,151 @@ import { focusStyles, highContrastOutlineStyles } from '../box';
-			borderCollapse: 'collapse',
-		},
-		'.rdp-tbody': {
-			border: 0,
+		'&:not([disabled]):hover': {
+			backgroundColor: boxPalette.backgroundShade,
+			color: boxPalette.foregroundText,
+			textDecoration: 'underline',
 		},
-		'.rdp-cell': {
-			width: cellSize,
-			height: cellSize,
-			padding: 0,
-			textAlign: 'center',
+		...focusStyles,
+	},
+	// Table
+	'.rdp-months': {
+		display: 'flex',
+		height: '23.5rem',
+	},
+	'.rdp-month': {
+		margin: `0 ${mapSpacing(1)}`,
+		'&:first-of-type': { marginLeft: 0 },
+		'&:last-of-type': { marginRight: 0 },
+	},
+	'.rdp-table': {
+		margin: 0,
+		maxWidth: `calc(${cellSize} * 7)`,
+		borderCollapse: 'collapse',
+	},
+	'.rdp-tbody': {
+		border: 0,
+	},
+	'.rdp-cell': {
+		width: cellSize,
+		height: cellSize,
+		padding: 0,
+		textAlign: 'center',
+	},
+	'.rdp-day_selected:not([disabled]), .rdp-day_selected:focus:not([disabled]), .rdp-day_selected:active:not([disabled]), .rdp-day_selected:hover:not([disabled]), .rdp-day_selected:hover:not([disabled])':
+		{
+			'&:not(:focus)': highContrastOutlineStyles,
+			backgroundColor: boxPalette.selected,
+			color: boxPalette.backgroundBody,
+			fontWeight: tokens.fontWeight.bold,
 		},
-		'.rdp-day_selected:not([disabled]), .rdp-day_selected:focus:not([disabled]), .rdp-day_selected:active:not([disabled]), .rdp-day_selected:hover:not([disabled]), .rdp-day_selected:hover:not([disabled])':
-			{
-				'&:not(:focus)': highContrastOutlineStyles,
-				backgroundColor: boxPalette.selected,
-				color: boxPalette.backgroundBody,
-				fontWeight: tokens.fontWeight.bold,
-			},
+} as const;
+
+export const reactDayRangePickerStyles = (dateRange?: {
+	from?: Date;
+	to?: Date;
+}) => {
+	const { from, to } = dateRange ?? {};
+	const startStyles = { borderRadius: '50% 0 0 50%' };
+	const endStyles = { borderRadius: '0 50% 50% 0' };
+	return {
+		// Middle of the date range
 		'.rdp-day_selected:not([disabled]).rdp-day_range_middle': {
 			backgroundColor: boxPalette.selectedMuted,
 			color: boxPalette.foregroundText,
-			fontWeight: tokens.fontWeight.bold,
 			borderRadius: 0,
 		},
-		// Selected days have more rounded borders in range mode
-		...(range && {
-			'.rdp-day_range_start': {
-				fontWeight: tokens.fontWeight.bold,
-				borderRadius: 0,
-				borderTopLeftRadius: '50%',
+		// Start day of date range
+		'.rdp-day_range_start:not(.rdp-day_range_end)': startStyles,
+		// End day of date range
+		'.rdp-day_range_end:not(.rdp-day_range_start)': endStyles,
+		// Start and end days of date range
+		'.rdp-day_range_start.rdp-day_range_end': {
+			borderRadius: '0',
+			...(from && {
 				borderBottomLeftRadius: '50%',
-			},
-			'.rdp-day_range_end:not(.rdp-day_range_start)': {
-				fontWeight: tokens.fontWeight.bold,
-				borderRadius: 0,
-				borderTopRightRadius: '50%',
+				borderTopLeftRadius: '50%',
+			}),
+			...(to && {
 				borderBottomRightRadius: '50%',
-			},
-		}),
-	}) as const;
+				borderTopRightRadius: '50%',
+			}),
+		},
+	};
+};
```

Comment:
This can be further simplified to:

```
export const reactDayRangePickerStyles = (dateRange?: {
	from?: Date;
	to?: Date;
}) => {
	const { from, to } = dateRange ?? {};
	const startStyles = {
		borderRadius: 0,
		borderBottomLeftRadius: '50%',
		borderTopLeftRadius: '50%',
	};
	const endStyles = {
		borderRadius: 0,
		borderBottomRightRadius: '50%',
		borderTopRightRadius: '50%',
	};

	return {
		// Middle of the date range
		'.rdp-day_selected:not([disabled]).rdp-day_range_middle': {
			backgroundColor: boxPalette.selectedMuted,
			color: boxPalette.foregroundText,
			borderRadius: 0,
		},
		// Start day of date range
		'.rdp-day_range_start:not(.rdp-day_range_end)': startStyles,
		// End day of date range
		'.rdp-day_range_end:not(.rdp-day_range_start)': endStyles,
		// Start and end days of date range
		'.rdp-day_range_start.rdp-day_range_end': {
			...(from && startStyles),
			...(to && endStyles),
		},
	};
};
```